File: 백준/Bronze/2139. 나는 너가 살아온 날을 알고 있다/나는 너가 살아온 날을 알고 있다.py
def is_leap_year(year):
    return (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0)

month_days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

while True:
    d, m, y = map(int, input().split())
    if d == 0 and m == 0 and y == 0:
        break

    days = 0
    for i in range(m - 1):
        days += month_days[i]

    if is_leap_year(y) and m > 2:
        days += 1

    days += d

    print(days)

File: 백준/Bronze/2139. 나는 너가 살아온 날을 알고 있다/나는 너가 살아온 날을 알고 있다.java
import java.util.Scanner;

public class Main {
    
    public static boolean isLeapYear(int year) {
        return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int[] monthDays = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };

        while (true) {
            int d = sc.nextInt(); // day
            int m = sc.nextInt(); // month
            int y = sc.nextInt(); // year

            if (d == 0 && m == 0 && y == 0) {
                break;
            }

            int days = 0;

            for (int i = 0; i < m - 1; i++) {
                days += monthDays[i];
            }

            if (isLeapYear(y) && m > 2) {
                days += 1;
            }

            days += d;

            System.out.println(days);
        }

        sc.close();
    }
}